src/main/java/ch/njol/skript/expressions/ExprBanData.java

```diff
@@ -0,0 +1,218 @@
+package ch.njol.skript.expressions;
+
+import ch.njol.skript.Skript;
+import ch.njol.skript.classes.Changer;
+import ch.njol.skript.doc.*;
+import ch.njol.skript.lang.Expression;
+import ch.njol.skript.lang.ExpressionType;
+import ch.njol.skript.lang.SkriptParser;
+import ch.njol.skript.lang.util.SimpleExpression;
+import ch.njol.skript.util.Timespan;
+import ch.njol.util.Kleenean;
+import com.destroystokyo.paper.profile.PlayerProfile;
+import io.papermc.paper.ban.BanListType;
+import org.bukkit.BanEntry;
+import org.bukkit.Bukkit;
+import org.bukkit.OfflinePlayer;
+import org.bukkit.event.Event;
+import org.jetbrains.annotations.Nullable;
+
+import java.net.InetAddress;
+import java.net.UnknownHostException;
+
+@Name("Ban Details")
+@Description("Returns data about a player or IP ban. " +
+	"The source can be any string, but it's usually the player's name. " +
+	"The expiration date won't return a value if the ban is permanent.")
+@Examples({"set {_player} to offlineplayer(\"Notch\", false)",
+	"set {_expiration} to the date {_player}'s ban expires",
+	"set {_time.left} to difference between now and {_expiration}",
+	"set {_reason} to the reason {_player} was banned",
+	"send \"There is %{_time.left}% before %{_player}% gets unbanned! They were banned for '%{_reason}%'\" to player"})
+@Since("INSERT_VERSION")
+@RequiredPlugins("Spigot 1.20.1+")
+public class ExprBanData extends SimpleExpression<Object> {
+
+	static {
+		if (Skript.methodExists(BanEntry.class, "remove"))
+			Skript.registerExpression(ExprBanData.class, Object.class, ExpressionType.SIMPLE,
+				"[the] date %offlineplayer/string% was banned",
```

Comment:
Syntax is not a good option, it can be very confusing for readers and might affect the parser. I suggest something like `ban date of ...` as Efnilite also suggested.

Reply:
This current one seems much better than ban x of y to me.
Is there a reason you think it's poor?

Reply:
`date "SkriptDev" was banned` seems off to me at first glance, it doesn't immediately gives you a hint that it's a ban date. IMO `ban date` is easier to quickly recognize it's a ban date.

Reply:
I think it'd be improved by adding `that`:
`the date that "SkriptDev" was banned`

It buries the lead by like 2 words which i don't really think is an issue, especially when it's so much more natural/fluid imo. `ban date of "SkriptDev"` feels much more stilted and awkward and something i wouldn't actually say irl.

I'm not opposed to having it in addition, but i definitely think the current syntax is higher quality

Reply:
so do i add another syntax for `ban date` or ..?